fix(dracut.sh): skip README for AMD microcode generation

aafeijoo-suse opened the pull request:
This file was added in https://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git/commit/amd-ucode/README?id=89ec6198f13d1007563ff87aae5de209e993be07 and it should be skipped.

## Checklist
- [X] I have tested it locally
- [ ] I have reviewed and updated any documentation if relevant
- [ ] I am providing new code and test(s) for it

Fixes #2541

champtar commented on dracut.sh:
```diff
@@ -2154,6 +2154,8 @@ if [[ $early_microcode == yes ]]; then
                 done
                 for i in $_fwdir/$_fw/$_src; do
                     [[ -e $i ]] || continue
+                    # skip README
```
```suggestion
                    # skip README*
```

aafeijoo-suse replied:
That's why I used `str_starts` with the full path instead of `std_ends`, to handle the different possible compressions. Do you think a `*` is really necessary in the previous comment?

champtar replied:
The `*` for me document that we are using `str_starts` because it's needed and not by accident.
Someone rewriting this code without context might think they only need to skip `README`

aafeijoo-suse committed:
fix(dracut.sh): skip README for AMD microcode generation

This file was added in https://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git/commit/amd-ucode/README?id=89ec6198f13d1007563ff87aae5de209e993be07
and it should be skipped.

Fixes #2541

classabbyamp commented on dracut.sh:
```diff
@@ -2154,6 +2154,8 @@ if [[ $early_microcode == yes ]]; then
                 for i in $_fwdir/$_fw/$_src; do
                     [[ -e $i ]] || continue
+                    # skip README{.xz,.zst,...}
+                    str_starts "$i" "$_fwdir/$_fw/README" && continue
```
this just puts a bandage on a symptom of the problem, instead of fixing the actual problem: the proper way would be to only `cat` files that match the glob `*.bin` (for amd) or `??-??-??` (for intel)
